src/ansys/openapi/common/_api_client.py

```diff
@@ -207,12 +207,16 @@ def __call_api(
         self.last_response = response_data
 
         return_data: Union[requests.Response, DeserializedType, None] = response_data
+        deserialized_response = None
         if _preload_content:
             _response_type = response_type
             if response_type_map is not None:
                 _response_type = response_type_map.get(response_data.status_code, None)
 
             return_data = self.deserialize(response_data, _response_type)
+            deserialized_response = return_data
```

Comment:
This is all a bit weird. We define a variable called `return_data`, which is a union of both the request response type and our deserialized type, and initially contains the raw request response.

We then maybe overwrite that with the deserialized response, and then write that to a different variable `deserialized_response`.

I realize that `return_data` needs to be the union of the types eventually, but it might be cleaner if that union happens closer to the end of the function.